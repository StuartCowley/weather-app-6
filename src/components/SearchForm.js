import PropTypes from "prop-types";

import "../styles/SearchForm.css";

function SearchForm({ searchText, setSearchText, handleCitySearch }) {
  const handleInputChange = (event) => {
    setSearchText(event.target.value);
  };

  return (
    <section className="search-form">
      <input
        type="text"
        className="search-form__input"
        placeholder="Please enter a location"
        value={searchText}
        onChange={handleInputChange}
      />
      <button
        type="submit"
        className="search-form__button"
        onClick={handleCitySearch}
      >
        Search
      </button>
    </section>
  );
}

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  handleCitySearch: PropTypes.func.isRequired,
};

export default SearchForm;
