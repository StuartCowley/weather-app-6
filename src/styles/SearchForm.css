.search-form {
  margin-top: 1.5rem;
}

.search-form__input {
  border: 2px solid #ddd;
  border-radius: 99px;
  padding: 1rem 1.5rem;
  background-color: #fff;
  min-width: 300px;
}

.search-form__button {
  border: 2px solid #153253;
  border-radius: 99px;
  padding: 1rem 1.5rem;
  font-weight: 700;
  color: #fff;
  background-color: #153253;
  cursor: pointer;
  margin-left: 1rem;
}
