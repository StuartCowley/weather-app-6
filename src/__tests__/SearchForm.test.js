import { render, screen } from "@testing-library/react";

import SearchForm from "../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "Manchester",
    setSearchText: jest.fn(),
    handleCitySearch: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        handleCitySearch={validProps.handleCitySearch}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the search button", () => {
    const { getByText } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        handleCitySearch={validProps.handleCitySearch}
      />
    );

    expect(getByText("Search")).toBeInTheDocument();
  });

  it("renders the right placeholder text", () => {
    render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        handleCitySearch={validProps.handleCitySearch}
      />
    );

    const inputNode = screen.getByPlaceholderText("Please enter a location");

    expect(inputNode).toBeInTheDocument();
  });
});
