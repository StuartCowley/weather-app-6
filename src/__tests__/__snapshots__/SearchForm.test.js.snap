// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <section
    class="search-form"
  >
    <input
      class="search-form__input"
      placeholder="Please enter a location"
      type="text"
      value="Manchester"
    />
    <button
      class="search-form__button"
      type="submit"
    >
      Search
    </button>
  </section>
</DocumentFragment>
`;
